fab/fabfile.py

```diff
@@ -463,9 +502,9 @@ def update_code(use_current_release=False):
 
     with cd(env.code_root if not use_current_release else env.code_current):
         sudo('git remote prune origin')
-        sudo('git fetch origin {}'.format(env.code_branch))
-        sudo('git checkout %(code_branch)s' % env)
-        sudo('git reset --hard origin/%(code_branch)s' % env)
+        sudo('git fetch origin --tags')
+        sudo('git checkout {}'.format(git_tag))
```

Comment:
Yup, you'll just be in a "detached HEAD state", where you can't really save any new commits you make until you check out a new branch.  You can actually check out several types of refs in  git - branches, remote branches (`$ git checkout origin/master`), tags, or commit hashes.

Related fun fact: you can do `set --code_branch=<ref thing>` with fabric for any of those things and I think it would just work - so we could have deployed a specific commit with `$ fab production awesome_deploy --set code_branch=7d6f1ed`.